runtime/common/src/main/kotlin/kotlinx/serialization/modules/SerialModuleImpl.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("UNCHECKED_CAST")
+
+package kotlinx.serialization.modules
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.isInstanceOf
+import kotlin.collections.set
+import kotlin.reflect.KClass
+
+private typealias SerializersMap = MutableMap<KClass<*>, KSerializer<*>>
+
+/**
+ * A default implementation of [SerialModule]
+ * which uses hash maps to store serializers associated with KClasses.
+ */
+@PublishedApi
+internal class SerialModuleImpl : SerialModule {
```

Comment:
Should be private

Reply:
It would be `internal` since it is used in builders